BrightFuturesTests/ResultTests.swift

```diff
@@ -95,55 +105,55 @@ class ResultTests: XCTestCase {
         XCTAssert(r.isFailure)
         XCTAssertEqual(r.error!.domain, "DivisionByZeroError")
     }
-    
-    func testFlatMapFutureSuccess() {
```

Comment:
It's not possible to implement this version of `flatMap` but it's a bit weird as well, because it couples the `Result` type directly to the `Future` type

We could replace it with a free function that takes a result, like `public func flatMap<T>(result: Result<T,NSError>, @noescape f: T -> Future<U>) -> Future<U>`, but it would be a breaking change.